src/mobu/handlers/github_ci_app.py

```diff
@@ -0,0 +1,150 @@
+"""Github webhook handlers for CI app."""
+
+import asyncio
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, HTTPException
+from gidgethub import routing
+from gidgethub.sansio import Event
+from safir.github.webhooks import GitHubCheckRunEventModel
+from safir.slack.webhook import SlackRouteErrorHandler
+
+from ..config import config
+from ..constants import GITHUB_WEBHOOK_WAIT_SECONDS
+from ..dependencies.context import RequestContext, anonymous_context_dependency
+from ..dependencies.github import (
+    ci_manager_dependency,
+    github_config_dependency,
+)
+from ..models.github import GitHubCheckSuiteEventModel, GitHubConfig
+from ..services.github_ci.ci_manager import CiManager
+
+__all__ = ["api_router"]
+
+api_router = APIRouter(route_class=SlackRouteErrorHandler)
+"""Registers incoming HTTP GitHub webhook requests"""
+
+
+gidgethub_router = routing.Router()
+"""Registers handlers for specific GitHub webhook payloads"""
+
+
+@api_router.post(
+    "/webhook",
+    summary="GitHub CI webhooks",
+    description="Receives webhook events from the GitHub mobu CI app.",
+    status_code=202,
+)
+async def post_webhook(
+    context: Annotated[RequestContext, Depends(anonymous_context_dependency)],
+    github_config: Annotated[GitHubConfig, Depends(github_config_dependency)],
+    ci_manager: Annotated[CiManager, Depends(ci_manager_dependency)],
+) -> None:
+    """Process GitHub webhook events for the mobu CI GitHubApp.
+
+    Rejects webhooks from organizations that are not explicitly allowed via the
+    mobu config. This should be exposed via a Gafaelfawr anonymous ingress.
+    """
+    webhook_secret = config.github_ci_app.webhook_secret
+    body = await context.request.body()
+    event = Event.from_http(
+        context.request.headers, body, secret=webhook_secret
+    )
+
+    owner = event.data.get("organization", {}).get("login")
+    if owner not in github_config.accepted_github_orgs:
+        context.logger.debug(
+            "Ignoring GitHub event for unaccepted org",
+            owner=owner,
+            accepted_orgs=github_config.accepted_github_orgs,
+        )
+        raise HTTPException(
+            status_code=403,
+            detail=(
+                "Mobu is not configured to accept webhooks from this GitHub"
+                " org."
+            ),
+        )
+
+    # Bind the X-GitHub-Delivery header to the logger context; this
+    # identifies the webhook request in GitHub's API and UI for
+    # diagnostics
+    context.rebind_logger(github_app="ci", github_delivery=event.delivery_id)
+    context.logger.debug("Received GitHub webhook", payload=event.data)
+    # Give GitHub some time to reach internal consistency.
+    await asyncio.sleep(GITHUB_WEBHOOK_WAIT_SECONDS)
```

Comment:
If there's a GitHub API we can poll to see when it's ready, we should probably do that rather than a fixed wait, but if not, sure, this'll work.

Reply:
This is a GitHub irritation :( From what I understand, there's no single thing we can poll for in all cases, we'd have to poll for the specific things that we want to work with to be in the state described in the webhook, or implement retries (like some other folks [have done](https://github.com/runatlantis/atlantis/issues/1019) with this problem).

* Since we don't know what GitHub resources we'll be working with at this point, we don't know yet what we should poll for
* We're using this janky waiting [elsewhere](https://github.com/lsst-sqre/times-square/blob/main/src/timessquare/handlers/external/endpoints.py#L98-L99), so at least this is consistent within our own stuff :)
* That's what's in the [GidgetHub docs](https://gidgethub.readthedocs.io/en/latest/#quickstart), for whatever that's worth...